components/brave_ads/browser/ads_service_factory.cc

```diff
@@ -31,9 +31,6 @@ namespace brave_ads {
 // static
 AdsService* AdsServiceFactory::GetForProfile(
     Profile* profile) {
-  if (profile->IsOffTheRecord())
```

Comment:
To return null service for OTR profiles, we could just use the implementation of BrowserContextKeyedServiceFactory::GetBrowserContextToUse which has the default behavior to return null services for OTR profiles.
https://cs.chromium.org/chromium/src/components/keyed_service/content/browser_context_keyed_service_factory.cc?dr=CSs&g=0&l=61-71